### DISCUSSION TITLE
Jinja2 Scripting

### DISCUSSION THREAD

**Discussion by TrixaFett** [netbox-community/netbox#17442] [Q&A]
I am trying to make a Script, that takes a specific Config template and add it to a device when i try to create it.
like this:
```
from dcim.models import Platform

def run(self, data, commit):
    platform = Platform.objects.get(name='Switch')
    for number in range(0, data['switch_no']):
            print(number)
            device = Device(
                platform=platform
             )
```

but for config_template
This Does not work
```
from dcim.models import ConfigTemplate

def run(self, data, commit):
        config_template = ConfigTemplate.objects.get(name='SwitchConf')
        for number in range(0, data['switch_no']):
            print(number)
            device = Device(
                config_template=config_template
            )
```

Can anyone help?

**Comment by candlerb** [comment]
Neither of those scripts is complete, as you've not doing

```
device.full_clean()
device.save()
```

(which I presume is what you're trying to do - create new Device objects?) nor are you providing the mandatory parameters to Device (i.e. role, device_type, site). Your imports are also wrong, e.g. 

```
>>> from dcim.models import ConfigTemplate
Traceback (most recent call last):
  File "/usr/lib/python3.12/code.py", line 90, in runcode
    exec(code, self.locals)
  File "<console>", line 1, in <module>
ImportError: cannot import name 'ConfigTemplate' from 'dcim.models' (/opt/netbox/netbox/dcim/models/__init__.py)
```

(at least in Netbox 4.0.11, it is `from extras.models import ConfigTemplate`)

So please start with a complete, runnable script which reproduces your problem, and then describe in what way it "does not work".

I can't see why assigning Device.config_template from a ConfigTemplate object instance won't work: the attribute comes from `RenderConfigMixin`
